frontend/hub/administration/repositories/RepositoryForm.tsx

```diff
@@ -0,0 +1,309 @@
+        />
+        <PageFormTextInput<RepositoryFormProps>
+          name="description"
+          label={t('Description')}
+          placeholder={t('Enter a repository description')}
+        />
+        <PageFormTextInput<RepositoryFormProps>
+          name="retain_repo_versions"
+          type="number"
+          min={1}
+          label={t('Retained number of versions')}
+          placeholder={t('Enter a number of versions')}
+        />
+        <PageFormGroup
+          label={t('Distributions')}
+          labelHelp={t(
+            'Content in repositories without a distribution will not be visible to clients for sync, download or search.'
+          )}
+        >
+          <PageFormWatch<string> watch={'name'}>
+            {(name: string) => {
+              return (
+                <PageFormCheckbox<RepositoryFormProps>
+                  name="createDistribution"
+                  label={t(`Create a "${name}" distribution`)}
+                  isDisabled={isDistributionDisabled}
+                />
+              );
+            }}
+          </PageFormWatch>
+        </PageFormGroup>
+        <PageFormGroup
+          label={t('Pipeline')}
+          labelHelp={t(
+            'Pipeline adds repository labels with pre-defined meanings: None - users require permissions to modify content in this repository to upload collection. Approved - collections can be moved here on approval. Publishing directly to this repository is disabled. Staging - collections uploaded here require approval before showing up on the search page. Anyone with upload permissions for a namespace can upload collections to this repository'
+          )}
+        >
+          <PageFormSelect
+            name={'pipeline'}
+            placeholderText={t('Select a pipeline')}
+            isRequired
+            options={[
+              { value: 'none', label: t('None') },
+              { value: 'approved', label: t('Approved') },
+              { value: 'staging', label: t('Staging') },
+            ]}
+          />
+        </PageFormGroup>
+        <PageFormGroup
+          label={t('Labels')}
+          labelHelp={t(
+            'Repository labels can change the context in which a repository is seen.\n' +
+              'Hide from search (hide_from_search) - prevent collections in this repository from showing up on the home page\n' +
+              '(pipeline: *) - see Pipeline above'
+          )}
+        >
+          {Object.keys(repositoryFormValues?.pulp_labels).map((label) => (
+            <Label key={label}>
+              {label}
+              {repositoryFormValues.pulp_labels[label] &&
+                `: ${repositoryFormValues.pulp_labels[label]}`}
+            </Label>
+          ))}
+          {Object.keys(repositoryFormValues?.pulp_labels).length === 0 && t('None')}
+          <br />
+          <PageFormWatch<string> watch={'pipeline'}>
+            {(pipeline: string) => {
+              return (
+                <HookWrapper>
```

Comment:
This `HookWrapper` could be moved to the framework. It might be helpful to others.

Reply:
This seems to be Hub specific edge case. I'd prefer to leave is as is.